elb_log_ingestor/main.py

```diff
@@ -66,26 +68,28 @@ def start_server():
         logs_to_be_processed, logs_processed, records, parser_stats
     )
     shipper = elasticsearch_shipper.ElasticsearchShipper(
-        #es_client, records, index_pattern, shipper_stats
-        None, records, index_pattern, shipper_stats
+        es_client, records, index_pattern, shipper_stats
     )
 
     # prepare the ApiEndpoint class for use
-    api_endpoint.ApiEndpoint.parser_stats = parser_stats 
+    api_endpoint.ApiEndpoint.parser_stats = parser_stats
     api_endpoint.ApiEndpoint.shipper_stats = shipper_stats
     api_endpoint.ApiEndpoint.fetcher = fetcher
     api_endpoint.ApiEndpoint.shipper = shipper
 
     server = http.server.HTTPServer(server_address, api_endpoint.ApiEndpoint)
 
-    fetcher_thread = threading.Thread(target=fetcher.run, daemon=True)
+    fetcher_thread = threading.Thread(target=fetcher.run, daemon=False)
     parser_thread = threading.Thread(target=parser.run, daemon=True)
-    shipper_thread = threading.Thread(target=shipper.run, daemon=True)
-    server_thread = threading.Thread(target=server.serve_forever)
+    shippers = []
+    for i in range(shipper_count):
+        shippers.append(threading.Thread(target=shipper.run, daemon=True))
+    server_thread = threading.Thread(target=server.serve_forever, daemon=True)
```

Comment:
this is the real magic to getting this going faster. Elasticsearch can take logs seemingly as fast as we can send them, but http overhead seems to be our biggest bottleneck. Cranking shipper threads up works around that